### PR TITLE
stop default for istio.io/waypoint-for in istioctl

### PR DESCRIPTION
**Pull request by ilrudie**
**Please provide a description of this PR:**

changes the behavior of istioctl waypoint generation/creation to avoid adding the istio.io/waypoint-for label if the user did not specify a value. istiod already defaults to service if the label is not present and if we explicitly set a value with istioctl then accepting the default from the gateway class in the future becomes more difficult.

### CODE REVIEW

**Comment by ilrudie** [comment]
cc @morrislaw, I believe you may have initially written this

**Review by howardjohn — APPROVED** [review]

**Review by linsun — APPROVED** [review]
Good one Ian!

**Review by MorrisLaw — APPROVED** [review]

File: istioctl/pkg/waypoint/waypoint.go [hunk 1]
```diff
@@ -86,16 +86,20 @@ func Cmd(ctx cli.Context) *cobra.Command {
...
-		// Determine which traffic address type to apply the waypoint to, if none is provided it will default to "service"
-		// as the waypoint-for traffic type.
-		if !validTrafficTypes.Contains(trafficType) {
-			return nil, fmt.Errorf("invalid traffic type: %s. Valid options are: %s", trafficType, validTrafficTypes.String())
-		}
 
-		if gw.Labels == nil {
-			gw.Labels = map[string]string{}
+		// only label if the user has provided their own value, otherwise we let istiod choose a default at runtime (service)
+		// this will allow for gateway class to provide a default for that class rather than always forcing service or requiring users to configure correctly
```
Comment (MorrisLaw) [inline istioctl/pkg/waypoint/waypoint.go:91]: This is a good idea!

**Pull request merged**